src/timeline-manager.tsx

```diff
@@ -0,0 +1,308 @@
+import {h, createRef} from 'preact';
+import * as KalturaPlayer from '@playkit-js/kaltura-player-js';
+// @ts-ignore
+import {CuePoint} from './components/cue-point';
+import {TimelineMarker} from './components/marker/timeline-marker';
+import {
+  Chapter,
+  QuizQuestionData,
+  TimedCuePointOptionsObject,
+  TimelineMarkerDataObject
+} from "../flow-typed/types/cue-point-option";
+import {SeekbarPreviewOptionsObject} from "../flow-typed/types/seekbar-preview-option";
+import {ThumbnailInfo, TimeLineMarker, TimelineMarkerProps} from "./types/timelineTypes";
+import {TimelinePreview} from "./components/marker/timeline-preview";
+
+const {preact, redux, reducers, style} = KalturaPlayer.ui;
+// @ts-ignore
+const {actions} = reducers.seekbar;
+// @ts-ignore
+const {core} = KalturaPlayer;
+const KalturaPlayerSeekBarSelector = '.playkit-seek-bar';
+const quizQuestionType = 'QuizQuestion';
+const chapterType = 'Chapter';
+
+/**
+ * @class TimelineManager
+ */
+class TimelineManager {
+  _uiManager: any;
+  _store: any;
+  _cuePointsRemoveMap: {[id: string]: Function};
+  _counter: number;
+  _cuePointsMap: Map<number, TimelineMarkerDataObject>;
+  _chapters: Chapter[] = [];
+
+  /**
+   * @constructor
+   * @param _player
+   * @param _logger
+   * @param _dispatchTimelineEvent
+   */
+  constructor(private _player: KalturaPlayerTypes.Player, private _logger: any, private _dispatchTimelineEvent: (event: string) => void) {
+    this._uiManager = this._player.ui;
+    this._store = redux.useStore();
+    this._cuePointsRemoveMap = {};
+    this._counter = 0;
+    this._cuePointsMap = new Map();
+  }
+
+  private _isNavigationPluginOpen = () => {
+    const navigationPlugin: any = this._player.getService('navigation');
+    if (!navigationPlugin) {
+      this._logger.warn("navigationPlugin haven't registered");
+      return false;
+    }
+    return navigationPlugin.isPluginActive();
+  };
+
+  private _isNavigationPluginVisible = () => {
+    const navigationPlugin: any = this._player.getService('navigation');
+    if (!navigationPlugin) {
+      this._logger.warn("navigationPlugin haven't registered");
+      return false;
+    }
+    return navigationPlugin.isVisible();
+  };
+
+  private _seekTo = (time: number) => {
+    this._player.currentTime = time;
+  };
+
+  private _handleChapter = (chapter: Chapter) => {
+    if (this._chapters.length > 0) {
+      // set the previous chapter's endTime to be the current chapter's startTime
+      this._chapters[this._chapters.length - 1].endTime = chapter.startTime;
```

Comment:
what happens if only 1 chapter in media?

Reply:
@semarche-kaltura the first chapter will not go inside this "if" block because i add it only after this check; see line 77.